### PR TITLE
Add copy pid button

### PR DESCRIPTION
**Pull request by svituz**
This PR adds a "Copy on clipboard" icon support to `string.vue` component and implements it for the PID attribute of Biobanks.
The icon button simply copies the value of the link to clipboard, so the user can, for example, share it.
In the case of the PID value in BiobankReport it copies the URL of the PID of the biobank.

#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- User documentation updated
- [x] Clean commits
- [x] No warnings during install
- [x] Added to release notes

### CODE REVIEW

File: src/components/generators/view-components/string.vue [hunk 1]
```diff
@@ -33,6 +57,13 @@ export default {
   methods: {
     displayName (item) {
       return item.label || item.name || item.id
+    },
+    copyOnClipboard (link, event) {
+      navigator.clipboard.writeText(link)
+      this.copyPidShown = true
+      setTimeout(() => {
```
Comment (jelmerveen) [inline src/components/generators/view-components/string.vue:64]: it is better to assign the setTimeOut to a variable and call clearTimeout when the function inside has ended (memory leak avoidance), in this case after this.copyPidShown assignment

File: src/components/generators/view-components/string.vue [hunk 2]
```diff
@@ -43,9 +74,21 @@ export default {
   transition: transform 0.1s;
   box-shadow: 0 0 0 1px white;
 }
+
 .badge:hover {
   transform: scale(1.4);
 }
+
+.fa-clipboard {
+  margin-left: 5px;
```
Comment (jelmerveen) [inline src/components/generators/view-components/string.vue:83]: can you use a bootstrap class? ml-1 or ml-2? if not, try to use rems

File: src/components/generators/view-components/string.vue [hunk 3]
```diff
@@ -54,4 +97,38 @@ export default {
 .fa-external-link:hover {
   cursor: pointer;
 }
+
+.toast-container {
+  display: block;
+  max-width: 350px;
```
Comment (jelmerveen) [inline src/components/generators/view-components/string.vue:103]: Using rems is better for responsiveness.

File: src/components/generators/view-components/string.vue [hunk 4]
```diff
@@ -54,4 +97,38 @@ export default {
...
+  backface-visibility: hidden;
+  -webkit-backface-visibility: hidden;
+  background-clip: padding-box;
+  z-index: 1;
+  border-radius: 0.25rem;
+}
+
+.toast-container-top-center {
+  position: fixed;
+  min-height: 200px;
```
Comment (jelmerveen) [inline src/components/generators/view-components/string.vue:113]: rems :)

File: src/components/generators/view-components/string.vue [hunk 5]
```diff
@@ -54,4 +97,38 @@ export default {
...
+  margin-top: 5px;
```
Comment (jelmerveen) [inline src/components/generators/view-components/string.vue:114]: bootstrap class please

File: src/components/generators/view-components/string.vue [hunk 6]
```diff
@@ -54,4 +97,38 @@ export default {
...
+  top: 0;
+}
+
+.toast-container .toast {
+    background-color: rgba(230, 242, 255, 0.85);
+    border-color: rgba(184, 218, 255, 0.85);
+    color: #004085;
```
Comment (jelmerveen) [inline src/components/generators/view-components/string.vue:121]: for colors, use the bootstrap utility classes, so it adheres to any custom theme as well

File: tests/unit/specs/components/generators/view-components/string.spec.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+      expect(wrapper.vm.copyPidShown).toBeFalsy()
+
+      await wrapper.find('#copy-icon').trigger('click')
+      expect(wrapper.vm.copyPidShown).toBeTruthy()
+      expect(wrapper.find('.toast-container').isVisible()).toBeTruthy()
+      expect(navigator.clipboard.writeText).toHaveBeenCalledWith(attribute.linkValue)
+
+      jest.runAllTimers()
+      expect(wrapper.vm.copyPidShown).toBeFalsy()
+      // expect(wrapper.find('.toast-container').isVisible()).toBeFalsy()
```
Comment (jelmerveen) [inline tests/unit/specs/components/generators/view-components/string.spec.js:84]: Unused code?

**Review by jelmerveen — CHANGES_REQUESTED** [review]

File: src/components/generators/view-components/string.vue (hunk 1 above)
Comment (svituz) [reply src/components/generators/view-components/string.vue:64]: Ok, I changed it

File: src/components/generators/view-components/string.vue (hunk 2 above)
Comment (svituz) [reply src/components/generators/view-components/string.vue:83]: used ml-1

File: src/components/generators/view-components/string.vue (hunk 3 above)
Comment (svituz) [reply src/components/generators/view-components/string.vue:103]: used 25rem

File: src/components/generators/view-components/string.vue (hunk 4 above)
Comment (svituz) [reply src/components/generators/view-components/string.vue:113]: It wasn't actually needed the min-height so I deleted it

File: src/components/generators/view-components/string.vue (hunk 6 above)
Comment (svituz) [reply src/components/generators/view-components/string.vue:121]: used alert-info class for the div

File: tests/unit/specs/components/generators/view-components/string.spec.js (hunk 7 above)
Comment (svituz) [reply tests/unit/specs/components/generators/view-components/string.spec.js:84]: yes, removed it

File: src/components/generators/view-components/string.vue [hunk 8]
```diff
@@ -14,12 +14,41 @@
           {{ attribute.value }}
         </span>
       </template>
+      <template v-if="attribute.linkValue">
+        <span
+          id="copy-icon"
+          @click.prevent="copyOnClipboard(attribute.linkValue, $event)"
```
Comment (jelmerveen) [inline src/components/generators/view-components/string.vue:20]: $event is passed as parameter, but not used in the function?

And on a sidenote, the working term should be copyToClipboard to be more syntactically correct ;)

**Review by jelmerveen — APPROVED** [review]

**Pull request merged**